src/main/java/com/esri/core/geometry/ogc/OGCConcreteGeometryCollection.java

```diff
@@ -501,4 +538,445 @@ public int hashCode() {
 		
 		return hash;
 	}
+	
+	@Override
+	public double distance(OGCGeometry another) {
+		if (this == another)
+			return isEmpty() ? Double.NaN : 0;
+
+		double minD = Double.NaN;
+		for (int i = 0, n = numGeometries(); i < n; ++i) {
+			double d = geometryN(i).distance(another);
+			if (d < minD || Double.isNaN(minD)) {
+				minD = d;
+				if (minD == 0) {
+					break;
+				}
+			}
+		}
+
+		return minD;
+	}
+	
+	//
+	//Relational operations
+	@Override
+	public boolean disjoint(OGCGeometry another) {
+		if (isEmpty() || another.isEmpty())
+			return true;
+
+		if (this == another)
+			return false;
+
+		//TODO: a simple envelope test
+		
+		OGCConcreteGeometryCollection flattened1 = flatten();
+		if (flattened1.isEmpty())
+			return true;
+		OGCConcreteGeometryCollection otherCol = new OGCConcreteGeometryCollection(another, esriSR);
+		OGCConcreteGeometryCollection flattened2 = otherCol.flatten();
+		if (flattened2.isEmpty())
+			return true;
+		
+		for (int i = 0, n1 = flattened1.numGeometries(); i < n1; ++i) {
+			OGCGeometry g1 = flattened1.geometryN(i);
+			for (int j = 0, n2 = flattened2.numGeometries(); j < n2; ++j) {
+				OGCGeometry g2 = flattened2.geometryN(i);
+				if (!g1.disjoint(g2))
+					return false;
+			}
+		}
+		
+		return true;
+	}
+	
+	@Override
+	public boolean contains(OGCGeometry another) {
+		if (isEmpty() || another.isEmpty())
+			return false;
+		
+		if (this == another)
+			return true;
+
+		//TODO: a simple envelope test
+		
+		OGCConcreteGeometryCollection flattened1 = flatten();
+		if (flattened1.isEmpty())
+			return true;
+		OGCConcreteGeometryCollection otherCol = new OGCConcreteGeometryCollection(another, esriSR);
+		OGCConcreteGeometryCollection flattened2 = otherCol.flatten();
+		if (flattened2.isEmpty())
+			return true;
+		
+		for (int i = 0, n2 = flattened2.numGeometries(); i < n2; ++i) {
+			OGCGeometry g2 = flattened2.geometryN(i);
+			boolean good = false;
+			for (int j = 0, n1 = flattened1.numGeometries(); j < n1; ++j) {
+				OGCGeometry g1 = flattened1.geometryN(i);
```

Comment:
This should be `geometryN(j)` not `..(i)`. 

Also, it may happen that neither `g1` fully contains a `g2`, but collectively a set of `g1`s contain all points in `g2`. Here is a test case:

```
public class TestOGCContains
{
    @Test
    public void testGeometryCollection()
    {
        assertContains("GEOMETRYCOLLECTION (POINT (0 0), LINESTRING (0 1, 5 1))", "GEOMETRYCOLLECTION (MULTIPOINT (0 0, 2 1))");
    }

    private void assertContains(String wkt, String otherWkt)
    {
        OGCGeometry geometry = OGCGeometry.fromText(wkt);
        OGCGeometry otherGeometry = OGCGeometry.fromText(otherWkt);
        assertTrue(geometry.contains(otherGeometry));
        assertTrue(otherGeometry.within(geometry));
    }
}
```

Reply:
@mbasmanova Maria, I've added this test and fixed this. Collection contains now uses another.difference(this).isEmpty().